docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/kubernetes.md

```diff
@@ -1,8 +1,81 @@
 ---
-title: "Kubernetes agents"
+title: "Running Dagster+ agents on Kubernetes"
 displayed_sidebar: "dagsterPlus"
 sidebar_position: 30
 sidebar_label: "Kubernetes"
 ---
 
-# Kubernetes agents
\ No newline at end of file
+# Running Dagster+ agents on Kubernetes
+
+This page provides instructions for running the [Dagster+ agent](dagster-plus/getting-started/whats-dagster-plus#Agents) on a [Kubernetes](https://kubernetes.io) cluster.
+
+## Installation
+
+
+### Prerequisites
+
+You'll need [a Kubernetes cluster. This can be a self-hosted Kubernetes cluster or a managed offering like [Amazon EKS](https://aws.amazon.com/eks/), [Azure AKS](https://azure.microsoft.com/en-us/products/kubernetes-service), or [Google GKE](https://cloud.google.com/kubernetes-engine).
+
+We recommend installing the Dagster+ agent using [Helm](https://helm.sh).
+
+## Step 1: Create a Kubernetes namespace
+
+```shell
+kubectl create namespace dagster-cloud
+```
+
+## Step 2: Create an agent token secret
+
+[Generate an agent token](dagster-plus/deployment/tokens) and set it as a Kubernetes secret:
+
+```shell
+kubectl --namespace dagster-cloud create secret generic dagster-cloud-agent-token --from-literal=DAGSTER_CLOUD_AGENT_TOKEN=<token>
+```
+
+## Step 3: Add the Dagster+ agent Helm chart repository
+
+```shell
+helm repo add dagster-cloud https://dagster-io.github.io/helm-user-cloud
+helm repo update
+```
+
+## Step 4: Install the Dagster+ agent Helm chart
+
+```shell
+helm --namespace dagster-cloud install agent --install dagster-cloud/dagster-cloud-agent
+```
+
+## Upgrading
+
+You can use Helm to do rolling upgrades of your Dagster+ agent
+
+```shell
+helm repo update
+helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --set dagsterCloudAgent.image.tag=latest
+```
+
+## Common configurations
+
+You can customize your Dagster+ agent using [Helm values](https://artifacthub.io/packages/helm/dagster-cloud/dagster-cloud-agent?modal=values). Some common configuration include
+
+### Configuring your agents to serve branch deployments
+
+[Branch deployments](dagster-plus/deployment/branch-deployments) are lightweight staging environments created for each code change. To configure your Dagster+ agent to manage them:
+
+```shell
+helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --set dagsterCloud.branchDeployments=true
+```
+
+### High availability configurations
+
+You can configure your Dagster+ agent to run with multiple replicas. Each will manage its own code servers and work will be load balanced across all replicas.
+
+```shell
+helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --set dagsterCloudAgent.replicas=2
+```
```

Comment:
I think it would be worth adding a small section to this guide with two pieces:

- How to confirm the agent is working? (navigate to deployments/agents tab in dagster+) 
- How to troubleshoot if the agent is not working:
  - Just the basic commands to do a kubectl describe / kubectl logs on the agent pod
  - Maybe a pointer to try pinging dagster+ from the pod to check for outbound network access

Reply:
I would also add next steps bread crumb to the full k8s config page.  A compelling breadcrumb might be: 

## Next Steps
- You may want to set default Kubernetes values for different projects, eg the default cpu and memory resource limit for a job. This setting, and other configuration options, are detailed [here](/todo)

Reply:
+1 to all these!

Reply:
I added some basic troubleshooting (just logs)

Turns out we don't install `curl` in our agent container image but I did come up with the kind of cute:

```shell
kubectl exec -it $(kubectl get pods -l deployment=agent -o jsonpath="{.items[0].metadata.name}") -- python -c "import requests; print(requests.get('https://dagster.plus/').status_code)"
```

I think it's too cute to included - I shopped around internally for some feedback and I'm not convinced this is something we propose all that often. The other one that came up was `dagster api grpc-health-check` but I think that's also probably too in the weeds for this guide.

For next steps, I'm sort of thinking it'd be nice to have almost like a...K8s agent config cookbook? I might borrow some of the folks who are more familiar with common usage patterns and we can slap something together.

Reply:
> For next steps, I'm sort of thinking it'd be nice to have almost like a...K8s agent config cookbook?

That would be *amazing* and exactly the kind of thing users are looking for.

Reply:
I scheduled something for a bunch of us to knock this out next week. Feels like a good thing to crowdsource.